### PR TITLE
Decoupling

### PR DESCRIPTION
**Pull request by kavics**
Discussion, arguing, screaming...

### CODE REVIEW

**Comment by tusmester** [comment]
Ok, we've talked through the current status of these files.

To be continued...

File: src/ContentRepository/ApplicationModel/ApplicationStorage.cs [hunk 1]
```diff
@@ -304,7 +305,7 @@ private static AppNode LoadApps(out List<string> appNames, out List<Application>
             using (new SystemAccount())
             {
                 //var result = nq.Execute();
-                var result = ContentQuery.Query(SafeQueries.TypeIs, QuerySettings.AdminSettings, typeof(Application).Name);
+                var result = ContentQuery_NEW.Query(SafeQueries.TypeIs, QuerySettings.AdminSettings, typeof(Application).Name);
```
Comment (kavics) [inline src/ContentRepository/ApplicationModel/ApplicationStorage.cs]: Unorthodox names indicate that they elements are semifinished and mean strong intention that they will be refactored.

**Comment by kavics** [comment]
@tusmester please review my last commit:: "New index commit logic..."

**Comment by kavics** [comment]
@tusmester: see this commit:  https://github.com/SenseNet/sensenet/pull/140/commits/102c44fdcf58d243083727f2bc4dbc6ecd1cdc1d
This is a good example of using the RepositoryBuilder class in an initalization hook.

**Comment by tusmester** [comment]
Cool, we can use this pattern.

**Pull request merged**